test/e2e/network/ingress.go

```diff
@@ -1114,17 +1114,35 @@ var _ = SIGDescribe("Ingress API", func() {
 		ginkgo.By("deleting")
 		err = ingClient.Delete(context.TODO(), createdIngress.Name, metav1.DeleteOptions{})
 		framework.ExpectNoError(err)
-		_, err = ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
-		framework.ExpectEqual(apierrors.IsNotFound(err), true, fmt.Sprintf("expected 404, got %#v", err))
+		ing, err := ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
+		// If ingress controller does not support finalizers, we expect a 404.  Otherwise we validate finalizer behavior.
+		if err != nil {
```

Comment:
Let's put the if err == nil case first.

Can we make this easier to read

```
ExpectFinalizer := func(ing *v1.Ingress) {
  framework.ExpectNotEqual(ing.DeletionTimestamp, nil, fmt.Sprint("expected deletionTimestamp, got nil"))
  framework.ExpectEqual(len(ing.Finalizers) > 0, true, fmt.Sprintf("expected finalizers on ingress, got none"))
}
...
```

use  `expectFinalizer(ing)` instead of copy/paste

Reply:
done